compendium/generated/quiz-w05-case-class-solurows-generated.tex

```diff
@@ -4,4 +4,4 @@
   \code|val p4 = Mutable()                      | & 4 & ~~\Large$\leadsto$~~ &  B & \code|Not found| \\
   \code|p2.moved(dx = 1) == Pt(1, 3)            | & 5 & ~~\Large$\leadsto$~~ &  A & \code|true| \\
   \code|p3.move(dy = 1) == new MutablePt(5, 7)  | & 6 & ~~\Large$\leadsto$~~ &  F & \code|false| \\
-  \code|p2 == p3                                | & 7 & ~~\Large$\leadsto$~~ &  E & \verb|warning: always false|
+  \code|p2 = p3                                 | & 7 & ~~\Large$\leadsto$~~ &  E & \verb|Found: ...; Required: ...|
```

Comment:
I changed it to an assignment attempt as `p2 == p3` returns `false`, but no warning or error, unlike Scala 2. But as you've written in another comment `"x" == 1` should give a compile error in Scala 3, so I don't really know why it isn't working.

Reply:
Hmm, this works in Scala 3 REPL:

```scala                                                                                    
scala> val x: Int = 1
val x: Int = 1

scala> val y: String = "abc"
val y: String = abc

scala> x == y                                                                                          
1 |x == y
  |^^^^^^
  |Values of types Int and String cannot be compared with == or !=
```
Still not sure why it isn't working with the classes as defined in this task..

Reply:
I think the compiler can check this for built-in types. If one wants error for self-made classes, then strict equality and derives equals has to be switched on; a lot more details here: https://dotty.epfl.ch/docs/reference/contextual/multiversal-equality.html
But I'm not sure if this is in scope for the course. So if you can tweak the exercises so that students understand that the get help with built-in types like String and Int when they use `==` and if the create their own classes the we should be careful when comparing different types. For now. ( Then maybe I'll get to this during lectures eventually...) Ok?

Reply:
```
scala> class X(val i: Int)
// defined class X

scala> class Y(val s: String)
// defined class Y

scala> X(2) == Y("hej")
val res0: Boolean = false

scala> import scala.language.strictEquality                                                                                                                                               

scala> X(2) == Y("hej")
1 |X(2) == Y("hej")
  |^^^^^^^^^^^^^^^^
  |Values of types X and Y cannot be compared with == or !=

```

Reply:
Would be good if you can illustrate something similar to above in new subtasks?

Reply:
I see! I can take a look at this tomorrow.

Reply:
:+1:

Reply:
I've taken a quick look at it, and I think it's best to remove the last equality check from this task and instead demonstrate it in the new task, as it becomes too cluttered to implement it in a nice way.

Also, since this is eventually out of scope, should the task be in `extrauppgifter` or `fördjupningsuppgifter`, or should it be kept in `grunduppgifter`? My plan is to show how to use strict equality by using `scala.language.strictEquality`.

Reply:
Makes sense to make CanEquals "fördjupningsuppgift" and if you like to create something here it is very welcome but not high prio. But a simple import strictEqualty exercise should be part of the basic tasks I think. Pls make something inline with my repl example above in a new task, if you agree?

Reply:
Sounds good, I'll prioritize getting a task made for `grunduppgifter` first.

Reply:
I've created a new task for equality checking in `grunduppgifter` ([0d282ee](https://github.com/lunduniversity/introprog/pull/548/commits/0d282eeb2dfb177b7b3552ce1e20fb5d66cedec1)). Is it okay, or do I need to change/improve something? I've assumed in the task that the students have an idea from the lectures why there is a difference in the equality checks.